express-notes-api/index.js

```diff
@@ -0,0 +1,99 @@
+const fs = require('fs');
+const express = require('express');
+const app = express();
+const data = require('./data');
+const notes = data.notes;
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  for (const property in notes) {
+    arr.push(notes[property]);
+  }
+  res.json(arr);
+  res.sendStatus(200);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (!id || id < 0 || Number.isInteger(id) === false) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (!notes[id]) {
+    res.status(404).json({ error: `can not find note with id ${id}` });
+  } else {
+    res.status(200).json(notes[id]);
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  const body = req.body;
+  const newNote = {};
+  if (!body.content) {
+    res.status(400).json({ error: 'content is a required field' });
+    return;
+  }
+  newNote.id = data.nextId;
+  newNote.content = body.content;
+  notes[newNote.id] = newNote;
+  data.nextId++;
+  const stringify = JSON.stringify(data, null, 2);
+  fs.writeFile('data.json', stringify, err => {
+    if (err) {
+      console.error(err);
+      res.status(500).json({ error: 'Unexpected error occurrred' });
+    } else {
+      res.status(201).json(newNote);
+    }
+  });
+});
+
+app.delete('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (!id || id < 0 || Number.isInteger(id) === false) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+    return;
+  } else if (!notes[id]) {
+    res.status(404).json({ error: `can not find note with id ${id}` });
```

Comment:
This likely needs a `return` here, otherwise the code below will run and a second response will be attempted to be sent which will result in an error.